ccm_web/client/src/utils/handleErrors.ts

```diff
@@ -52,31 +59,31 @@ const handleErrors = async (resp: Response): Promise<void> => {
   if (resp.ok) return
   const text = await resp.text()
   console.error(text)
-  const errorBody: APIErrorData = JSON.parse(text)
+  let errorBody: APIErrorData
+  try {
+    errorBody = JSON.parse(text)
+  } catch (error) {
+    throw new Error(`Non-JSON error encountered with status code ${resp.status}`)
+  }
+
+  if (resp.status === 401 && errorBody.redirect === true) {
+    redirect('/')
```

Comment:
Does redirect to root, will make LTI launch happen?

Reply:
It won't be an LTI launch. It's just a redirect to Home page.

Reply:
Redirecting to root will basically reload the application, and then when `globals` shows they have no canvas token, they are shown the authorize prompt. We could change this logic somewhat and redirect to a new authorize page directly.